Introduce lombok and @data for entites and dtos

gabichelsea opened the pull request:
Trello card link:

https://trello.com/c/eBofE73X

gabichelsea committed:
lombok is defined as a maven dependency for parent pom and added persistence layer's pom and service layer's pom

@Data
@NoArgsConstructor
@AllArgsConstructor

annotations on many entities and DTOs and request objects

gabichelsea committed:
update README.md

gabichelsea committed:
fix it typo

gabichelsea committed:
Fix it
Use @Builder annotation
Change @Date annotation where we have to change

gabichelsea committed:
Fix it @Builder annotation problem

kbertalan commented on pom.xml:
```diff
@@ -20,7 +20,7 @@
         <environment>integration</environment>
         <com.epam.rft>1.0-SNAPSHOT</com.epam.rft>
-        <maven.war.plugin.version>2.3</maven.war.plugin.version>
+        <maven.war.plugin.version>2.2</maven.war.plugin.version>
```
There is no reason to change the war plugin version for this task

gabichelsea committed:
Finished:
Added @Builder annotation to SearchOptions and FilterRequest classes
Added @AllArgsConstructor annotation every classes where is was necessary after added @Builder annotation

Fix war plugin version number
Replace constructor + setter calls with builders
